src/main/java/gregtech/api/fluids/definition/MaterialFluidDefinition.java

```diff
@@ -0,0 +1,216 @@
+     * @param state            the state for the fluid
+     * @param tags             the tags for the fluid
+     * @param translationKey   the translation key for the fluid
+     * @param stillIconType    the icon type for the fluid when still
+     * @param flowingIconType  the icon type for the fluid when flowing
+     * @param color            the color of the fluid
+     * @param temperature      the temperature of the fluid in kelvin
+     * @param hasBlock         if the fluid has a block
+     * @param hasCustomTexture if the fluid has a custom texture
+     * @see Builder
+     */
+    public MaterialFluidDefinition(@Nonnull FluidType type, @Nonnull FluidState state, @Nonnull Collection<FluidTag> tags,
+                                   @Nonnull String translationKey, @Nonnull MaterialIconType stillIconType,
+                                   @Nonnull MaterialIconType flowingIconType, int color, int temperature,
+                                   boolean hasBlock, boolean hasCustomTexture) {
+        super(state, tags, translationKey, null, null, color, temperature, hasBlock);
+        this.type = type;
+        this.stillIconType = stillIconType;
+        this.flowingIconType = flowingIconType;
+        if (temperature == -1) { // override the super class's inference, handled elsewhere
+            this.temperature = -1;
+        }
+        this.hasCustomTexture = hasCustomTexture;
+    }
+
+    /**
+     * Construct a fluid from this definition
+     *
+     * @param material  the material to use
+     * @param fluidName the name of the fluid
+     * @return the fluid for the material
+     */
+    @Nonnull
+    public AdvancedMaterialFluid constructFluid(@Nonnull Material material, @Nonnull String fluidName) {
+        if (this.hasCustomTexture) {
+            ResourceLocation textureLocation = new ResourceLocation(GTValues.MODID, "blocks/fluids/fluid." + material);
+            this.setStill(textureLocation);
+            this.setFlowing(textureLocation);
+        } else {
+            final MaterialIconSet iconSet = material.getMaterialIconSet();
+            this.setStill(this.stillIconType.getBlockTexturePath(iconSet));
+            this.setFlowing(this.flowingIconType.getBlockTexturePath(iconSet));
+        }
+
+        if (this.temperature == -1) {
+            BlastProperty property = material.getProperty(PropertyKey.BLAST);
+            if (property == null) {
+                if (material.hasProperty(PropertyKey.DUST) && state != FluidState.PLASMA) {
+                    this.temperature = 1200;
+                } else {
+                    this.temperature = getInferredTemperature();
+                }
+            } else {
+                if (this.state == FluidState.LIQUID) this.temperature = property.getBlastTemperature();
+                else if (this.state == FluidState.PLASMA) this.temperature = 30_000 + property.getBlastTemperature();
+                else this.temperature = 298;
```

Comment:
All these temperature numbers should be somewhere as constant. So we can easily find where there are used and change them easily (not probable).